device/bluetooth_mapper/watcher/watcher.go

```diff
@@ -100,8 +101,22 @@ func onPeripheralDisconnected(p gatt.Peripheral, err error) {
 //onPeripheralConnected contains the operations to be performed as soon as the peripheral device is connected
 func (w *Watcher) onPeripheralConnected(p gatt.Peripheral, err error) {
 	actionmanager.GattPeripheral = p
-	DeviceConnected <- true
 	helper.ChangeDeviceState("online", deviceID)
+	ss, err := p.DiscoverServices(nil)
+	if err != nil {
+		glog.Errorf("Failed to discover services, err: %s\n", err)
+		os.Exit(1)
```

Comment:
right now our mapper manages only one device at a time right ? if we support multiple devices , then the watcher should not exit if it fails to connect to one device.  are we tracking future mapper improvements via open issues ?
please create issues for all such improvements.

Reply:
we have an open umbrella issue to track future improvements in the mapper. We can add support for multiple devices as one item in it